src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -154,13 +58,13 @@ public class TagHelperDescriptor
         /// <remarks>
         /// <c>*</c> at the end of an attribute name acts as a prefix match.
         /// </remarks>
-        public IReadOnlyList<string> RequiredAttributes { get; }
+        public IEnumerable<string> RequiredAttributes { get; [param: NotNull] set; } = Enumerable.Empty<string>();
 
         /// <summary>
         /// HTML element tag names targeted by tag helpers allowed as children.
         /// </summary>
         /// <remarks><c>null</c> indicates all tag helper children are allowed.</remarks>
-        public IReadOnlyList<string> AllowedChildren { get; }
+        public IEnumerable<string> AllowedChildren { get; set; }
```

Comment:
Not sure what you mean here. Previously there was no constructor that accepted `List<T>` directly, it was always `IEnumerable<T>`. We created a list out of that and set it to the read only list. I made this change because we agreed that the `IReadOnlyList<T>` was unnecessary and there was no other change needed anywhere else except in a test here https://github.com/aspnet/Razor/blob/taghelper-descriptor-fix/test/Microsoft.AspNet.Razor.Test/TagHelpers/TagHelperDescriptorTest.cs#L387.